addon/components/infinity-loader.js

```diff
@@ -111,45 +70,33 @@ const InfinityLoaderComponent = Ember.Component.extend({
     ) {
       return false;
     }
-    return this._bottomOfScrollableOffset() > this._triggerOffset();
-  },
 
-  _loadMoreIfNeeded() {
-    const infinityModel = this.get('infinityModelContent');
-    if (this._shouldLoadMore() && infinityModel) {
-      this.sendAction('loadMoreAction', infinityModel);
-    }
+    this._debounceScrolledToBottom();
   },
 
-  _setupScrollable() {
-    var scrollable = this.get('scrollable');
-    if (Ember.typeOf(scrollable) === 'string') {
-      var items = Ember.$(scrollable);
-      if (items.length === 1) {
-        this.set('_scrollable', items.eq(0));
-      } else if (items.length > 1) {
-        throw new Ember.Error("Ember Infinity: Multiple scrollable elements found for: " + scrollable);
-      } else {
-        throw new Ember.Error("Ember Infinity: No scrollable element found for: " + scrollable);
-      }
-      this.set('_customScrollableIsDefined', true);
-    } else if (scrollable === undefined || scrollable === null) {
-      this.set('_scrollable', Ember.$(window));
-      this.set('_customScrollableIsDefined', false);
-    } else {
-      throw new Ember.Error("Ember Infinity: Scrollable must either be a css selector string or left empty to default to window");
-    }
+  /**
+   * https://github.com/DockYard/ember-in-viewport#didenterviewport-didexitviewport
+   * 
+   * @method didExitViewport
+   */
+  didExitViewport() {
+    this._cancelTimers();
   },
 
-  loadedStatusDidChange: Ember.observer('infinityModelContent.reachedInfinity', 'destroyOnInfinity', function () {
-    if (this.get('infinityModelContent.reachedInfinity') && this.get('destroyOnInfinity')) {
-      this.destroy();
+  _debounceScrolledToBottom() {
+    /*
+     This debounce is needed when there is not enough delay between onScrolledToBottom calls.
+     Without this debounce, all rows will be rendered causing immense performance problems
+     */
+    function loadMore() {
+      this.sendAction('loadMoreAction', this.get('infinityModelContent'));
     }
-  }),
+    this._debounceTimer = run.debounce(this, loadMore, this.get('eventDebounce'));
+  },
```

Comment:
I'm not sure I understand this.  Is the issue that the `loadMore` call happens multiple times before the loader is pushed out of view?

Reply:
Yep.  So, for example, rAF will poll ~16ms, so some sort of debounce is needed.